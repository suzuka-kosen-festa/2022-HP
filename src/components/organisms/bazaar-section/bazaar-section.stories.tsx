import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from 'react';
import { BazaarSection } from ".";

type T = typeof BazaarSection;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  attentions: [
    "学内における飲食物の食べ歩きはご遠慮ください。",
    "学内バザーにて購入した飲食物のお持ち帰りはご遠慮ください",
    "各出店バザーにお並びいただく際、1つの出店につきお並びいただく人数は5人までとしておりますので、当日は係員の指示に従ってください。",
  ],
  attentionTitle: "Bazaar",
  buttonText: "バザーの詳細を見る",
  description: "飲食団体は18団体、その他レクリエーションには17団体が出店しています！屋台・レク等のバザー詳細についてはこちらをクリック！",
  title: "Bazaar"
};

export default {
  args,
  argTypes: {},
  component: BazaarSection,
} as Meta;

export const Default: Story = {};
