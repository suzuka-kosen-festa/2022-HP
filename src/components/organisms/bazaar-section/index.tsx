import type { FC } from "react";
import tw from "twin.macro";
import { Link } from "wouter";
import type { BazaarSectionProperties } from "./types/model";
import { Caution } from "../../molecules/caution";
import { Heading } from "../../atoms/heading";
import { Text } from "../../atoms/text";
import { Button } from "../../atoms/button";

const BazaarSectionContainer = tw.section`flex flex-col justify-center items-center`;

const BazaarSection: FC<BazaarSectionProperties> = ({
  attentionTitle,
  attentions,
  buttonText,
  description,
  title,
  ...rest
}) => (
  <BazaarSectionContainer aria-label="bazaar" {...rest}>
    <Heading css={tw`mb-4 sm:mb-6`} colorKey="purple">
      {title}
    </Heading>
    <Caution title={attentionTitle} sentence={attentions} css={tw`mb-4 sm:mb-8`} />
    <Text css={tw`whitespace-pre-wrap mb-6 sm:mb-16`}>{description}</Text>
    <Link to="/bazaar">
      <Button colorKey="purple" as="a" role="button">
        {buttonText}
      </Button>
    </Link>
  </BazaarSectionContainer>
);

export { BazaarSection };
