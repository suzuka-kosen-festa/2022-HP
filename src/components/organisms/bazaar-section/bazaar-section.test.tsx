import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./bazaar-section.stories";

const { Default } = composeStories(stories);

describe("(components) organisms/bazaar-section", () => {
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
