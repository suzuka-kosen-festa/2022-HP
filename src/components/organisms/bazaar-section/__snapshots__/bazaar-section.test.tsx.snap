// Vitest Snapshot v1

exports[`(components) organisms/bazaar-section > take snap shot 1`] = `
<div>
  <section
    aria-label="bazaar"
    class="css-wkpw2c"
  >
    <h1
      class="css-wtcl1w-BazaarSection"
    >
      Bazaar
    </h1>
    <section
      aria-label="注意事項"
      class="css-1kphsmg-BazaarSection"
      role="alert"
    >
      <div
        class="css-1m7spai"
      >
        <div
          class="css-1vlcokg"
        >
          <span />
        </div>
        <h2
          class="css-qt9rcz-Caution"
        >
          バザーに関するお願い
        </h2>
      </div>
      <ul
        class="css-mvofyw-Caution"
      >
        <li
          class="css-u80qbw-Caution"
        >
          学内における飲食物の食べ歩きはご遠慮ください。
        </li>
        <li
          class="css-u80qbw-Caution"
        >
          学内バザーにて購入した飲食物のお持ち帰りはご遠慮ください
        </li>
        <li
          class="css-u80qbw-Caution"
        >
          各出店バザーにお並びいただく際、1つの出店につきお並びいただく人数は5人までとしておりますので、当日は係員の指示に従ってください。
        </li>
      </ul>
    </section>
    <p
      class="css-sr7z0r-BazaarSection"
    >
      飲食団体は18団体、その他レクリエーションには17団体が出店しています！屋台・レク等のバザー詳細についてはこちらをクリック！
    </p>
    <a
      class="css-t2w93t-Button"
      href="/bazaar"
      role="button"
      tabindex="0"
    >
      <p
        class="css-oca8th-Button"
      >
        バザーの詳細を見る
      </p>
    </a>
  </section>
</div>
`;
